lib/hanami/model/config/adapter.rb

```diff
@@ -103,13 +103,10 @@ def instantiate_adapter(mapper)
           begin
             klass = Hanami::Utils::Class.load!(class_name, Hanami::Model::Adapters)
             klass.new(mapper, uri, options)
-          rescue NameError
-            raise AdapterNotFound.new(class_name)
           rescue => e
-            raise "Cannot instantiate adapter of #{klass} (#{e.message})"
+            raise Error, e
```

Comment:
This preserves only the message from the original exception, not the location.

Instead we bubble the error up, casting it as a `Hanami::Model::Error`